homeassistant/components/qnap_qsw/config_flow.py

```diff
@@ -63,3 +67,27 @@ async def async_step_user(
             ),
```

Comment:
You'll need to change the  line in async_step_user so it can win over a discovery `                await self.async_set_unique_id(format_mac(mac), raise_on_progress=False)`

Reply:
This shouldn't be needed after d9e7d8a86e74ed5203f14a0ba4a92f28714aa6cf

Reply:
You'll still need to do this because without `raise_on_progress=False` in `async_step_user` it will hit the below code will abort in the middle of the user trying to set it up when its also been discovered since the discovered flow is in in-progress flow.

https://github.com/home-assistant/core/blob/046d7d2a239e0eab370ff9d11b1885ee9c69dffe/homeassistant/config_entries.py#L1297

Reply:
Fixed in b9632f12ffde5c5c6b7ae5321ee872d368bd9c7a
Thanks @bdraco